File: blueprints/search/app/web/handler/api/knowledge.go
package api

import (
	"github.com/go-mizu/mizu"
	"github.com/go-mizu/mizu/blueprints/search/store/postgres"
)

// KnowledgeHandler handles knowledge panel requests
type KnowledgeHandler struct {
	store *postgres.Store
}

// NewKnowledgeHandler creates a new knowledge handler
func NewKnowledgeHandler(store *postgres.Store) *KnowledgeHandler {
	return &KnowledgeHandler{store: store}
}

// GetEntity returns a knowledge panel for an entity
func (h *KnowledgeHandler) GetEntity(c *mizu.Ctx) error {
	query := c.Param("query")
	if query == "" {
		return c.JSON(400, map[string]string{"error": "query parameter required"})
	}

	panel, err := h.store.Knowledge().GetEntity(c.Context(), query)
	if err != nil {
		return c.JSON(500, map[string]string{"error": err.Error()})
	}

	if panel == nil {
		return c.JSON(404, map[string]string{"error": "entity not found"})
	}

	return c.JSON(200, panel)
}

File: blueprints/search/app/worker/src/routes/knowledge.ts
import { Hono } from 'hono'
import { KnowledgeService } from '../services/knowledge'
import { CacheStore } from '../store/cache'
import type { HonoEnv } from '../types'

const app = new Hono<HonoEnv>()

app.get('/:query', async (c) => {
  const query = c.req.param('query')
  if (!query) {
    return c.json({ error: 'Missing required parameter: query' }, 400)
  }

  const cache = new CacheStore(c.env.SEARCH_KV)
  const knowledgeService = new KnowledgeService(cache)
  const panel = await knowledgeService.getPanel(query)

  if (!panel) {
    return c.json({ error: 'No knowledge panel found' }, 404)
  }

  return c.json(panel)
})

export default app
